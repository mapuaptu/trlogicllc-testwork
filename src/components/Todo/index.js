import Icon from '@/components/Icon/index.vue';

export default {
  name: 'Todo',
  components: {
    Icon,
  },
  props: {
    value: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    onInput({ target }) {
      return this.$emit('input', target.checked);
    },
  },
};
