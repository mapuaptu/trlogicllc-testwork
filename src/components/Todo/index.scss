@import '@/styles/_variables.scss';

.todo {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-radius: 5px;
  padding: 10px;
  min-width: 30px;
  background-color: $color-java;

  &:not(:last-child) {
    margin-bottom: 20px;
  }

  .main {
    display: flex;
    align-items: center;
  }

  .complete {
    margin-right: 20px;

    input {
      width: 20px;
      height: 20px;
      cursor: pointer;
    }
  }

  .content {
    margin-right: 20px;
    font-weight: 700;

    &--complete {
      text-decoration: line-through;
    }
  }

  .controls {
    display: flex;
    align-items: center;

    > div:not(:last-child) {
      margin-right: 10px;
    }
  }

  &--complete {
    opacity: 0.6;
  }
}
