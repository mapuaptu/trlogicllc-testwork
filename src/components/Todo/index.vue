<template>
  <div :class="[$style.todo, value && $style['todo--complete']]">
    <div :class="$style.main">
      <div :class="$style.complete">
        <input
          type="checkbox"
          :checked="value"
          :disabled="disabled"
          @input="onInput"
        >
      </div>

      <div :class="[$style.content, value && $style['content--complete']]">
        <slot />
      </div>
    </div>

    <div
      v-if="!disabled"
      :class="$style.controls"
    >
      <Icon icon="pencil-outline" />
      <Icon icon="delete-outline" />
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
