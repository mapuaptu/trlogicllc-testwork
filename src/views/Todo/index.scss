@import '@/styles/_variables.scss';

.todo {
  position: relative;
}
