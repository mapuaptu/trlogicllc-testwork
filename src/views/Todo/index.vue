<template>
  <div :class="$style.todo">
    TODO
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
