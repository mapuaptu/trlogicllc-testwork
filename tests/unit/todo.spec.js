import Todo from '@/components/Todo/index.vue';
import { mount } from '@vue/test-utils';

describe('Note component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Todo);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
