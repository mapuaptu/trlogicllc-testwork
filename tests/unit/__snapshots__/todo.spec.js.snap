// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component 1. Rendered success 1`] = `
<div class="">
  <div>
    <div><input type="checkbox"></div>
    <div class=""></div>
  </div>
  <div>
    <div><svg width="24" height="24" viewbox="0 0 24 24">
        <path d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z" fill="#fff"></path>
      </svg></div>
    <div><svg width="24" height="24" viewbox="0 0 24 24">
        <path d="M6,19A2,2 0 0,0 8,21H16A2,2 0 0,0 18,19V7H6V19M8,9H16V19H8V9M15.5,4L14.5,3H9.5L8.5,4H5V6H19V4H15.5Z" fill="#fff"></path>
      </svg></div>
  </div>
</div>
`;
